lib/azure/armrest/storage_account_service.rb

```diff
@@ -146,7 +197,11 @@ def list_private_images(group = configuration.resource_group)
 
         list(group).each do |lstorage_account|
           threads << Thread.new(lstorage_account) do |storage_account|
-            key = list_account_keys(storage_account.name, group).fetch('key1')
+            if Time.parse(api_version).utc >= Time.parse('2016-01-01').utc
+              key = list_account_key_objects(storage_account.name, group).first.key
+            else
+              key = list_account_keys(storage_account.name, group).fetch('key1')
```

Comment:
Yes, did some local testing, but feel free to double check.

The .key method just returns a string. So, `.fetch('key1')` and `.key` are basically the same. The only difference is that the `.key` method will use key1 if present, and fall back to key2 if not present, so it's slightly more robust.